docs/writing-tests/test-runner-with-vitest.mdx

```diff
@@ -0,0 +1,502 @@
+        headless: true,
+      },
+      // Speed up tests and better match how they run in Storybook itself
+      // https://vitest.dev/config/#isolate
+      // Consider removing this if you have flaky tests
+      isolate: false,
+      setupFiles: ['./.storybook/vitest.setup.ts'],
+    },
+  })
+);
+```
+</details>
+
+<details>
+  <summary>Example Vitest workspace file</summary>
+
+  If you're using a [Vitest workspace](https://vitest.dev/guide/workspace), you can define a new workspace project:
+
+```ts title="vitest.workspace.ts"
+import { defineWorkspace } from 'vitest/config';
+import { storybookTest } from '@storybook/experimental-addon-test/vite-plugin';
+
+export default defineWorkspace([
+  // This is the path to your existing Vitest config file
+  './vitest.config.ts', 
+  {
+    name: 'storybook',
+    // This is the path to your existing Vite config file
+    extends: './vite.config.ts',
+    plugins: [
+      storybookTest({
+        storybookScript: 'yarn storybook --ci',
+      }),
+    ],
+    test: {
+      // Glob pattern to find story files
+      include: ['src/**/*.stories.?(m)[jt]s?(x)'],
+      // Enable browser mode
+      browser: {
+        enabled: true,
+        name: 'chromium',
+        // Make sure to install Playwright
+        provider: 'playwright',
+        headless: true,
+      },
+      // Speed up tests and better match how they run in Storybook itself
+      // https://vitest.dev/config/#isolate
+      // Consider removing this if you have flaky tests
+      isolate: false,
+      setupFiles: ['./.storybook/vitest.setup.ts'],
+    },
+  }
+]);
+```
+</details>
+
+## Usage
+
+There are three primary ways to run tests using the Vitest plugin:
+
+### CLI
+
+The plugin transforms your stories into real Vitest tests, so you run those tests just like you run any other Vitest tests in your project. Typically, you will have a `test` script in your `package.json` that runs your tests.
+
+If you don't already have a `test` script, you can add one that runs Vitest:
+
+```json title="package.json"
+{
+  "scripts": {
+    "test": "vitest"
+  }
+}
+```
+
+If you already have a `test` script that runs something other than Vitest, you can either adjust it to run Vitest (as above) or add a new script that runs Vitest:
+
+```json title="package.json"
+{
+  "scripts": {
+    "test-storybook": "vitest"
+  }
+}
+```
+
+When you run that script, the addon will find and run your story-based tests. Here's an example of running your tests (in [watch mode](https://vitest.dev/guide/cli.html#vitest-watch), by default) using the Vitest CLI:
+
+<CodeSnippets path="addon-test-run-tests.md" />
+
+### Editor extension
+
+Transforming your stories into Vitest tests with the plugin also enables you to run and debug tests using Vitest [IDE integrations](https://vitest.dev/guide/ide.html). This allows you to run tests directly from your editor, such as VSCode and JetBrains IDE.
+
+[TK - Screenshot of VS Code]
```

Comment:
I can assist but I think @jonniebigodes knows better on making consistent screenshots throughout the docs?

The feature is about showing the sidebar and check icons on the story file:
<img width="802" alt="image" src="https://github.com/user-attachments/assets/85d3d453-86e0-4ade-90b3-696b57fd883d">

This appears when you use the [vscode extension](https://vitest.dev/guide/ide.html) and set up the plugin